_posts/2021-10-01-how-to-pre-assignments.md

```diff
@@ -0,0 +1,67 @@
+- Technology choices. List chosen technologies. It's also nice to know why you chose those technologies.
+- TODO. If there are some things missing or not working, you can list them in README.
+
+## Git History
+Your git history is also something that we will often look into before digging deeper into your code. From your git history we can get a glimpse of your development process: where you started, in which order you did things, which parts were refactored during the process, etc. Knowing how to use version control (nowadays analogous to git) is one of the most essential skills of a developer. To make your project look more professional you should pay some attention to how your git history looks. 
+
+During the Dev Academy recruitment process we saw many git histories that had basically two commits. One containing all the application code and the second updating readme. Please try to have more commits than that. A good rule of thumb is to have one commit per one major change in your code. A good commit message describes the changes your commit makes to the repository. Try to be concise but yet descriptive, not easy, we know :) Try to think what is the one main thing you did in this commit, if there are several, then maybe that is an indication that you should have several commits instead of one.
+
+Some people like to keep they git history extra neat by amending and squashing their commits. In a recruitment pre-assignment it is usually enough to just commit frequently enough and in reasonably sized chunks.
+
+Examples of good commits:
+![Git history](/img/pre-assignments/git-history.png)
+
+## Make sure that code compiles and tests pass
+Make sure that the instructions on how to run the project actually work. The reviewer checking out your project does not want to fix stuff to make it work. If you have tests, they should all pass or at least you should have some explanation for failing tests. 
+
+## Focus on getting features complete
+Pre-assignment can consist of many requirements and features. Some of them can be required and some optional. For this fall’s Solita Dev Academy pre-assignment we didn't have any must dos and the applicant's could choose what they wanted to focus on.
+
+If you think that you can't finish everything, it's better to do some things thoroughly and leave the others undone than to do everything half-way. Choose your focus points and what you want to showcase, then start with those and do other things afterwards.
+
+## Write good quality code
+Good quality code is easy to read and looks nice. For instance make sure that you always use the same indentation across your project and remove any unused variables and functions. Use linters and code formatters to automatise this process, because otherwise you will miss a lot of stuff. These tools can visually highlight problems in your code and you can enable format on save, which can for instance add missing semicolons and remove extra white space.
```

Comment:
for instance, make sure...
to automate
otherwise, you will miss ...
on saving/when saving/upon saving